### ISSUE TITLE
SQLite & dotnet fm - Could not load file or assembly

### ISSUE THREAD

**Issue by glufsaren** [fluentmigrator/fluentmigrator#863]
Hello,

I'm trying to use the FluentMigrator.DotNet.Cli tool.

I've added it to the project file (tested both on .NET Standard 2.0 and .NET Core 2.0)
`<ItemGroup><DotNetCliToolReference Include="FluentMigrator.DotNet.Cli" version="2.0.5" /></ItemGroup>`

If I run:
`dotnet fm migrate --processor sqlserver`
everything works as expected, but if I run:
`dotnet fm migrate --processor sqlite`
I get an exeption:

```
System.Exception: Failed to load file C:\Projects\Strieker\Strieker.Test.UI\bin\debug\netcoreapp2.0\Microsoft.Data.Sqlite.dll ---> System.IO.FileNotFoundException: Could not load file or assembly 'C:\Projects\Strieker\Strieker.Test.UI\bin\debug\netcoreapp2.0\Microsoft.Data.Sqlite.dll'. The system cannot find the file specified.
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromPath(IntPtr ptrNativeAssemblyLoadContext, String ilPath, String niPath, ObjectHandleOnStack retAssembly)
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String assemblyPath)
   at System.Reflection.Assembly.LoadFile(String path)
   at FluentMigrator.Runner.Processors.ReflectionBasedDbFactory.TryLoadAssemblyFromDirectories(IEnumerable`1 directories, String assemblyName, ICollection`1 exceptions, Assembly& assembly) in Z:\buildAgent\work\89282cfb76b17ca5\src\FluentMigrator.Runner.Core\Processors\ReflectionBasedDbFactory.cs:line 224
   --- End of inner exception stack trace --- (Could not load file or assembly 'System.Data.SQLite, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.) (Could not load file or assembly 'System.Data.SQLite, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.)
``` 

If I run the migrations via code using `var factory = new SQLiteProcessorFactory();` everything works as well. Perhaps it has to do with the context in which dotnet-fm is run? 

I'm running the dotnet fm from the project directory and I've referenced:

```xml
    <PackageReference Include="FluentMigrator" Version="2.0.5" />
    <PackageReference Include="FluentMigrator.Console" Version="2.0.5" />
    <PackageReference Include="FluentMigrator.Runner" Version="2.0.5" />
    <PackageReference Include="FluentMigrator.Runner.SQLite" Version="2.0.5" />
    <PackageReference Include="Microsoft.Data.Sqlite" Version="2.0.1" />
    <PackageReference Include="Microsoft.Data.Sqlite.Core" Version="2.0.1" />
    <PackageReference Include="System.Data.SQLite" Version="1.0.108" />
```

Please let me know if you need any further information

**Comment by fubar-coder** [comment]
The problem is indeed that the the `dotnet-fm` tool runs in a different than your project/assembly, which means that it doesn't have access to the assemblies you're referencing.

You may try the following steps:

1. `dotnet publish` your project
2. Add the `FluentMigrator.DotNet.Cli` package as tool to your project (`<DotNetCliToolReference Include="FluentMigrator.DotNet.Cli" Version="2.0.5" />` (skip if already done)
3. `dotnet restore` (skip if step 2 skipped)
4. From your project directory (i.e. where your `csproj` is), Execute `dotnet fm migrate` with the appropriate options (e.g. a full path to your published assembly). You can use `dotnet fm --help`  and `dotnet fm migrate --help` to see all options.

Partially fixed after #857 is done.

**Comment by glufsaren** [comment]
Thanks! 😃

**Comment by fubar-coder** [comment]
Fixed in develop branch, .NET Core 2.1 preview 2 tooling is required, will be in 3.0.0

**Comment by giggio** [comment]
Cool, I'll pick the alphas from https://www.myget.org/F/fluent-migrator/api/v3/index.json until 3.0 is out.
